javatests/arcs/android/storage/ttl/PeriodicCleanupTaskTest.kt

```diff
@@ -0,0 +1,96 @@
+import arcs.core.entity.awaitReady
+import arcs.core.host.EntityHandleManager
+import arcs.core.storage.api.DriverAndKeyConfigurator
+import arcs.core.storage.keys.DatabaseStorageKey
+import arcs.core.storage.referencemode.ReferenceModeStorageKey
+import arcs.jvm.host.JvmSchedulerProvider
+import arcs.jvm.util.testutil.FakeTime
+import com.google.common.truth.Truth.assertThat
+import kotlin.coroutines.EmptyCoroutineContext
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@Suppress("EXPERIMENTAL_API_USAGE", "UNCHECKED_CAST")
+@RunWith(AndroidJUnit4::class)
+class PeriodicCleanupTaskTest {
+    private val collectionKey = ReferenceModeStorageKey(
+        backingKey = DatabaseStorageKey.Persistent("backing", DummyEntity.SCHEMA_HASH),
+        storageKey = DatabaseStorageKey.Persistent("collection", DummyEntity.SCHEMA_HASH)
+    )
+    private val fakeTime = FakeTime()
+    private lateinit var worker: PeriodicCleanupTask
+
+    @Before
+    fun setUp() {
+        val context : Context = ApplicationProvider.getApplicationContext()
+        DriverAndKeyConfigurator.configure(AndroidSqliteDatabaseManager(context))
+        SchemaRegistry.register(DummyEntity.SCHEMA)
+        worker = TestWorkerBuilder.from(context, PeriodicCleanupTask::class.java).build()
+    }
+
+    @Test
+    fun ttlWorkerTest() = runBlocking {
+        // Set time in the past so entity is already expired.
+        fakeTime.millis = 1L
+
+        val handle = createCollectionHandle()
+        val entity = DummyEntity().apply { num = 1.0 }
+        handle.storeAndWait(entity)
+        withContext(handle.dispatcher) {
+            assertThat(handle.fetchAll()).containsExactly(entity)
+        }
+
+        // Trigger worker.
+        assertThat(worker.doWork()).isEqualTo(Result.success())
+
+        // Verify entity is gone.
+        withContext(handle.dispatcher) {
+            assertThat(handle.fetchAll()).isEmpty()
+        }
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    private suspend fun createCollectionHandle() =
+        EntityHandleManager(
+            time = fakeTime,
+            scheduler = JvmSchedulerProvider(EmptyCoroutineContext)("test")
+        ).createHandle(
+            HandleSpec(
+                "name",
+                HandleMode.ReadWrite,
+                CollectionType(EntityType(DummyEntity.SCHEMA)),
+                DummyEntity
+            ),
+            collectionKey,
+            Ttl.Days(2)
+        ).awaitReady() as ReadWriteCollectionHandle<DummyEntity>
+
+    private suspend fun ReadWriteCollectionHandle<DummyEntity>.storeAndWait(entity: DummyEntity) {
+        val deferred = CompletableDeferred<Unit>()
+        onUpdate { deferred.complete(Unit) }
+        runBlocking(dispatcher) {
+            store(entity).join()
+        }
+        deferred.await()
```

Comment:
If this is to test the database deletion, it woudl probably be more robust to add a listener to the databaseimpl which completes a Job when it sees the entity you're looking for...  otherwise it's possible this particular block could finish only after the store receives the operation, but the writeback hasn't flushed it to disk yet.

Reply:
Added a wait on the WriteBack (or is there another better way to wait for the database?)
I checked for flakiness by going in the copybara client and successfully running
blaze test third_party/java_src/arcs/javatests/arcs/android/storage/database:PeriodicCleanupTaskTest --runs_per_test=5000 --jobs=5000